src/mcmc/hmc.jl

```diff
@@ -1,7 +1,46 @@
-abstract type Hamiltonian <: InferenceAlgorithm end
+# AbstractSampler interface for Turing
+
+abstract type Hamiltonian <: AbstractMCMC.AbstractSampler end
```

Comment:
I want to tag @yebai for visibility.

I can see the motivation for `InferenceAlgorithm`, but I think this makes things a lot cleaner. Because we can just dispatch on `Hamiltonian` instead of `Sampler{<:Hamiltonian}`.